sqlmesh/core/renderer.py

```diff
@@ -249,6 +249,11 @@ def _resolve_table(
             dialect=self._dialect,
             copy=False,
         )
+        # We quote the table here to mimic the behavior of _resolve_tables, otherwise we may end
+        # up normalizing twice, because _to_table_mapping returns the mapped names unquoted.
```

Comment:
Is there any test we can come up with which results in unexpected behavior if the table was normalized twice?

Reply:
Perhaps we could have a test where `@this_model` is rendered in the context of a Snowflake model and the schema becomes `"sqlmesh_FOO"`. Without this fix we'd get `"SQLMESH_FOO"`.

Reply:
(I'm planning to implement the test soon)

Reply:
Aight I gave it a whirl, see https://github.com/TobikoData/sqlmesh/pull/3494/commits/5c05c4d73cf798527b0d8b7158813aed2783fecb